internal/pkg/mining/worker.go

```diff
@@ -208,32 +204,31 @@ func (w *DefaultWorker) Mine(ctx context.Context, base block.TipSet, nullBlkCoun
 		outCh <- Output{Err: err}
 		return
 	}
-	// lookback consensus.ElectionLookback for the election ticket
-	baseHeight, err := base.Height()
+	baseEpoch, err := base.Height()
 	if err != nil {
 		log.Warnf("Worker.Mine couldn't read base height %s", err)
 		outCh <- Output{Err: err}
 		return
 	}
-	ancestors, err := w.getAncestors(ctx, base, baseHeight+(abi.ChainEpoch(nullBlkCount+1)))
-	if err != nil {
-		log.Warnf("Worker.Mine couldn't get ancestorst %s", err)
-		outCh <- Output{Err: err}
-		return
-	}
-	electionTicket, err := sampling.SampleNthTicket(int(miner.ElectionLookback-1), ancestors)
+	// Look-back for the election ticket.
+	// The parameter is interpreted as: lookback=1 means parent tipset. Subtract one here because the base from
+	// which the lookback is counted is already the parent, rather than "current" tipset.
+	// The sampling code will handle this underflowing past the genesis.
+	targetEpoch := baseEpoch - (miner.ElectionLookback - 1) + abi.ChainEpoch(nullBlkCount)
+	entropy, err := MakeElectionPoStChallengeSeedEntropy(targetEpoch, w.minerAddr)
```

Comment:
epoch is used as part of samplinig randomness below now, no need for it in entropy here